Add C# version of the project

skyace65 opened the pull request:
Adds a C# version of the project using code from https://github.com/godotengine/godot-docs/pull/4724. Closes https://github.com/godotengine/godot-docs/issues/4427.

Also during the process of doing this I found a small error in the code, paulloz is going to open a PR in the docs to fix it, but the fix is in this PR so it's good to merge.

skyace65 committed:
Merge branch 'C#Version' of https://github.com/skyace65/godot-3d-dodge-the-creeps into C#Version

mhilbrunner commented:
cc @NathanLovato (and maybe @paulloz or @aaronfranke in case they can look over the C#)

aaronfranke commented on dodge_the_creeps_3d_c#/Main.cs:
```diff
@@ -0,0 +1,44 @@
+
+	public override void _Ready()
+	{
+		GD.Randomize();
```
As per the [C# style guide](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_style_guide.html), use 4 spaces instead of tabs to follow the C# language standard.

aaronfranke commented on dodge_the_creeps_3d_c#/.gitignore:
```diff
@@ -0,0 +1,2 @@
+.import
+logs/
```
This file is not necessary (and actually should be deleted from the GDScript version too).

NathanLovato commented:
I'll leave the C# review to others, I'm not the right person to assess if the code's idiomatic and whatnot.

skyace65 commented:
I've converted everything from tabs to spaces, and removed the .gitignore file from the regular and C# version. @paulloz I'm unable to recreate your issue.

raulsntos commented on dodge_the_creeps_3d_c#/Mob.cs:
```diff
@@ -0,0 +1,47 @@
+
+public class Mob : KinematicBody
+{
+    // Emitted when the played jumped on the mob.
```
```suggestion
    // Emitted when the player jumped on the mob.
```

skyace65 commented:
Fixed.

akien-mga commented:
Thanks for the contribution! This project is archived and replaced by https://github.com/godotengine/godot-demo-projects/tree/master/3d/squash_the_creeps, please contribute there instead!
